### PR TITLE
Realised nad's issue for removing an instance of module ⊆-Reasoning. 

### PR DESCRIPTION
**Pull request by alhassy**
Just trying to contribute; please c.f. #310.

### CODE REVIEW

File: src/Data/List/Relation/Sublist/Extensional/Propositional/Properties.agda [hunk 1]
```diff
@@ -36,45 +36,11 @@ private
...
-  ⊆-reflexive refl = id
-
-  ⊆-refl : Reflexive (_⊆_ {A = A})
-  ⊆-refl x∈xs = x∈xs
-
-  ⊆-trans : Transitive (_⊆_ {A = A})
-  ⊆-trans xs⊆ys ys⊆zs x∈xs = ys⊆zs (xs⊆ys x∈xs)
+-- Relational properties and Reasoning over subsets
 
+open import Data.List.Relation.Sublist.Extensional.Setoid.Properties using (module OrderDefinitions)
```
Comment (MatthewDaggitt) [inline src/Data/List/Relation/Sublist/Extensional/Propositional/Properties.agda]: This should go at the top of the file.

**Comment by MatthewDaggitt** [comment]
> Just trying to contribute;

Thank you very much! We really appreciate it. 

So one problem with simply using the same module for both is that the propositional reasoning will expose the `_≈⟨_⟩_` combinator which is both unnecessary and confusing as it's really the `_≡⟨_⟩_` combinator in disguise. It looks like this problem already exists now, but better to fix it while we can! The best idea is probably to open the setoid reasoning and hide the unnecessary combinators.

I'm happy to merge this in, but just be aware that issue #185 also comes into play as the mechanics underlying preorder/partial order reasoning are also going to change soon.

**Comment by MatthewDaggitt** [comment]
Actually what we need to do is:

- In the setoid reasoning rename `_≈⟨_⟩_` to `_≋⟨_⟩_`
- In the propositional reasoning hide `_≋⟨_⟩_`.

**Comment by MatthewDaggitt** [comment]
Just had a play with this as I've never used it myself. I'm struggling to come up with an example of when `_∈⟨_⟩_` operator is useful. It can't be chained with the other operators and it's not compatible with `begin`. Anyone have any examples using it?

It might work better if it was in the `Membership` modules inside its own `Reasoning` module...

**Comment by gallais** [comment]
I suspect it's meant to be used *instead* of `begin_`.

**Comment by MatthewDaggitt** [comment]
> @MatthewDaggitt I cannot relocate _∈⟨_⟩_ to Data.List.Membership.Setoid, if that's what you meant, since that would create a cyclic dependency.

Okay, that's fine, leave it in for now.

> Is there anything else I should do to have this request accepted?

So I've just noticed that the proposed changes break backwards compatibility unnecessarily as the `A : Set a` parameter will now need to be given explicitly for the refl, sym, trans proofs in the `Propositional` file, whereas before it was implicit. To be honest I think for such a small number of proofs the `OrderDefinitions` module is probably overkill. I'd be more tempted to just restate the proofs in the `Propositional` file and instantiate them each directly. It's much clearer to users what is and what isn't exported by the file then.

Also a CHANGELOG entry would be much appreciated!

**Comment by MatthewDaggitt** [comment]
Merging in and will tweak. Thanks for the PR.

**Pull request merged**